src/main/java/com/cs203/smucode/controllers/NotificationController.java

```diff
@@ -10,44 +11,87 @@
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
 
 @RestController
 @RequestMapping("/notifications")
 public class NotificationController {
 
+    private static final Logger logger = LoggerFactory.getLogger(NotificationController.class);
+
     private final NotificationMapper notificationMapper;
     private final INotificationService notificationService;
+    private final EventHandler eventHandler;
 
     @Autowired
     public NotificationController(INotificationService notificationService,
-                                  NotificationMapper notificationMapper) {
+                                  NotificationMapper notificationMapper, EventHandler eventHandler) {
         this.notificationService = notificationService;
         this.notificationMapper = notificationMapper;
+        this.eventHandler = eventHandler;
     }
 
-    @PostMapping("/")
+    /**
+     * Endpoint to handle incoming notifications from event services.
+     *
+     * <p>This method receives a notification payload, validates and converts it,
+     * and then triggers any associated event handling logic.
+     * It persists the notification and sends it to relevant subscribers in real-time.
+     * </p>
+     *
+     * @param notificationDTO The notification DTO received from
+     *                        the event-sending service, containing details such
+     *                        as message type, recipient information, and content.
+     * @return ResponseEntity containing the created Notification object if successful,
+     *         along with an HTTP 200 OK status.
+     *
+     * @throws ApiRequestException if the notification type is invalid or
+     *         if any error occurs during notification creation or processing.
+     */
+    @PostMapping("/stream")
     public ResponseEntity<Notification> createNotification(
         @RequestBody @Valid NotificationDTO notificationDTO
     ) {
         try {
-            return ResponseEntity.ok(
-                    notificationService.createNotification(
-                            notificationMapper.notificationDTOtoNotification(notificationDTO)
-                    )
-            );
+            Notification notification = notificationMapper.notificationDTOtoNotification(notificationDTO);
+            notificationService.createNotification(notification);
+
+            logger.info("Notification created: {}", notification);
+            // Handle incoming event, eg. notify subscribed users
+            eventHandler.handleEvent(notification);
+            return ResponseEntity.ok(notification);
+
         } catch (IllegalArgumentException e) {
             throw new ApiRequestException("Invalid notification type");
         } catch (Exception e) {
-            throw new ApiRequestException("Something went wrong creating a notification");
+            throw new ApiRequestException("Something went wrong creating a notification; " + e.getMessage());
```

Comment:
dont do e.get message, as we dont know what the exception is, it may propagate sensitive matter to our frontend, this sensitive info can be like about our db schema (like sqlstate violated), or auth failed things.
use logger.error("e.getmessage", e)

Reply:
okay

Reply:
agree with this